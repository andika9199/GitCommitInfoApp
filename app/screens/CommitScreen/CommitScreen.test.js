import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import ComponentConnect, {Commit, mapDispatchToProps} from './index';

const mockStore = configureMockStore();
const initialProps = {
  commitList: {
    data: [
      {
        commit: {
          message: 'Always return an EventDispatcher in bridgeless mode...',
          committer: {
            date: '2020-05-27T19:16:21Z',
          },
        },
        author: {
          login: 'ejanzer',
          avatar_url: 'https://avatars1.githubusercontent.com/u/2308395?v=4',
        },
        committer: {
          login: 'facebook-bot',
          avatar_url: 'https://avatars3.githubusercontent.com/u/6422482?v=4',
        },
      },
      {
        commit: {
          message: 'Always return an EventDispatcher in bridgeless mode...',
          committer: {
            date: '2020-05-27T19:16:21Z',
          },
        },
        author: {
          login: 'charles',
          avatar_url: 'https://avatars1.githubusercontent.com/u/2308395?v=4',
        },
        committer: {
          login: 'facebook-bot',
          avatar_url: 'https://avatars3.githubusercontent.com/u/6422482?v=4',
        },
      },
    ],
    error: false,
    isFetching: false,
  },
  getRepoCommitList: jest.fn(),
  route: {
    params: {
      repo: 'facebook-react-native',
    },
  },
  navigation: {
    reset: jest.fn(),
  },
};

describe('Commit Screen', () => {
  describe('Commit Screen rendered', () => {
    it('Screen Renderer Successfully', async () => {
      const component = renderer.create(<Commit {...initialProps} />);
      expect(component).toMatchSnapshot();
    });
  });
  describe('Commit Screen function', () => {
    const component = shallow(<Commit {...initialProps} />);
    const instance = component.instance();

    it('newerCommit', async () => {
      instance.setState({
        page: 1,
      });
      await instance.newerCommit();
      expect(instance.props.getRepoCommitList).toBeCalled();
      expect(instance.state.page).toBe(0);
    });
    it('olderCommit', async () => {
      instance.setState({
        page: 1,
      });
      await instance.olderCommit();
      expect(instance.props.getRepoCommitList).toBeCalled();
      expect(instance.state.page).toBe(2);
    });
    it('logOut', async () => {
      await instance.logOut();
      expect(instance.props.navigation.reset).toBeCalled();
    });
  });
  describe('Redux', () => {
    const initialState = {
      commit: {
        data: {},
        error: false,
        isFetching: false,
      },
    };
    const store = mockStore(initialState);
    const component = shallow(<ComponentConnect store={store} />);

    it('mapStateToProps commitList should return true', () => {
      expect(component.props().commitList).not.toEqual({
        data: {},
        isFetching: false,
        error: false,
      });
    });
    it('mapDispatchToProps getRepoCommitList call right type', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).getRepoCommitList();
      expect(dispatch.mock.calls[0][0]).toEqual({
        payload: {
          keyword: undefined,
          page: undefined,
        },
        type: 'GET_COMMIT_LIST',
      });
    });
  });
});
