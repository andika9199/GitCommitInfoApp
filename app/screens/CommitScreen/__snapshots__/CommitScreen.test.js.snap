// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit Screen Commit Screen rendered Screen Renderer Successfully 1`] = `
<RCTScrollView>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#333333",
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Robot",
            "fontSize": 13,
            "fontWeight": "bold",
          }
        }
      >
        logout
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#cccccc",
            "borderWidth": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 4,
          },
          Object {
            "backgroundColor": "#cccccc",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#333333",
            "fontFamily": "Robot",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Always return an EventDispatcher in bridgeless mode...
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginTop": 2,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://avatars1.githubusercontent.com/u/2308395?v=4",
              }
            }
            style={
              Object {
                "height": 20,
                "resizeMode": "contain",
                "width": 20,
              }
            }
          />
          <Image
            source={
              Object {
                "uri": "https://avatars3.githubusercontent.com/u/6422482?v=4",
              }
            }
            style={
              Object {
                "height": 20,
                "resizeMode": "contain",
                "width": 20,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "paddingLeft": 4,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "fontFamily": "Robot",
                "fontSize": 13,
              }
            }
          >
            ejanzer authored and facebook-bot commited a day ago
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#cccccc",
            "borderWidth": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 4,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#333333",
            "fontFamily": "Robot",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Always return an EventDispatcher in bridgeless mode...
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginTop": 2,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://avatars1.githubusercontent.com/u/2308395?v=4",
              }
            }
            style={
              Object {
                "height": 20,
                "resizeMode": "contain",
                "width": 20,
              }
            }
          />
          <Image
            source={
              Object {
                "uri": "https://avatars3.githubusercontent.com/u/6422482?v=4",
              }
            }
            style={
              Object {
                "height": 20,
                "resizeMode": "contain",
                "width": 20,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "paddingLeft": 4,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "fontFamily": "Robot",
                "fontSize": 13,
              }
            }
          >
            charles authored and facebook-bot commited a day ago
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginVertical": 16,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#cccccc",
            "borderRadius": 4,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "Robot",
              "fontSize": 13,
              "fontWeight": "bold",
            }
          }
        >
          Newer
        </Text>
      </View>
      <View
        style={
          Object {
            "width": 20,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#cccccc",
            "borderRadius": 4,
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
            "width": 150,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "Robot",
              "fontSize": 13,
              "fontWeight": "bold",
            }
          }
        >
          Older
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
